src/utils/EquipmentType.js

```diff
@@ -172,56 +173,65 @@ const sortEquipments = (a, b) => {
         : a.label.localeCompare(b.label);
 };
 
-export const renderEquipmentForSearchBar = (classes, intl) => {
-    return (props, element, { inputValue }) => {
-        let matches = match(element.label, inputValue, {
-            insideWords: true,
-            findAllOccurrences: true,
-        });
-        let parts = parse(element.label, matches);
-        /* override li.key otherwise it will use label which could be duplicated */
+export const TagRenderer = ({ props, element }) => {
+    if (
+        element.type !== EQUIPMENT_TYPE.SUBSTATION.name &&
+        element.type !== EQUIPMENT_TYPE.VOLTAGE_LEVEL.name
+    )
         return (
-            <li {...props} key={element.key}>
-                <div className={classes.equipmentOption}>
-                    <span
-                        className={clsx(
-                            classes.equipmentTag,
-                            classes.equipmentTypeTag
-                        )}
-                    >
-                        {intl.formatMessage({
-                            id: EQUIPMENT_TYPE[element.type].tagLabel,
-                        })}
-                    </span>
-                    <OverflowableText
-                        text={parts.map((e) => e.text).join()}
-                        className={classes.result}
-                    >
-                        {parts.map((part, index) => (
-                            <span
-                                key={index}
-                                style={{
-                                    fontWeight: part.highlight
-                                        ? 'bold'
-                                        : 'inherit',
-                                }}
-                            >
-                                {part.text}
-                            </span>
-                        ))}
-                    </OverflowableText>
-                    {element.type !== EQUIPMENT_TYPE.SUBSTATION.name &&
-                        element.type !== EQUIPMENT_TYPE.VOLTAGE_LEVEL.name && (
-                            <OverflowableText
-                                text={element.voltageLevelLabel}
-                                className={clsx(
-                                    classes.equipmentTag,
-                                    classes.equipmentVlTag
-                                )}
-                            />
-                        )}
-                </div>
-            </li>
+            <OverflowableText
+                text={element.voltageLevelLabel}
+                className={clsx(
+                    props.classes.equipmentTag,
+                    props.classes.equipmentVlTag
+                )}
+            />
         );
-    };
+    return <></>;
+};
+
+export const RenderEquipmentForSearchBar = ({
+    inputValue,
+    tagRenderer = TagRenderer,
```

Comment:
We could use a TagRenderer by default but maybe we should use a more generic name for the component prop.
tagRenderer -> suffixRenderer ? I don't know.
It seems like this is a very generic component with 3 parts. Each part could have a proper renderer but mainly contains text or buttons. It's a pity to make it too specialized I think.